dll/win32/msgina/msgina.c

```diff
@@ -161,6 +161,55 @@ ChooseGinaUI(VOID)
     HeapFree(GetProcessHeap(), 0, SystemStartOptions);
 }
 
+static BOOL
+SafeGetUnicodeString(const LSA_UNICODE_STRING *pInput, PWSTR pszOutput, SIZE_T cchMax)
+{
+    if (pInput && pszOutput)
+    {
+        PWSTR pszInput = pInput->Buffer;
+        SIZE_T cchInput = pInput->Length / sizeof(WCHAR);
+        if (cchInput < cchMax && (pszInput || !cchInput))
+        {
+            CopyMemory(pszOutput, pszInput, cchInput * sizeof(WCHAR));
+            pszOutput[cchInput] = UNICODE_NULL;
+            return TRUE;
+        }
+    }
+    if (pszOutput && cchMax)
+        *pszOutput = UNICODE_NULL;
+    return FALSE;
+}
+
+static BOOL
+GetLsaDefaultPassword(PGINA_CONTEXT pgContext)
+{
+    // learn.microsoft.com/en-us/windows/win32/secauthn/protecting-the-automatic-logon-password
+
```

Comment:
Reference link before the function definition... but maybe even just use SAL2?
```suggestion
/* Reference: https://learn.microsoft.com/en-us/windows/win32/secauthn/protecting-the-automatic-logon-password */
static BOOL
GetLsaDefaultPassword(PGINA_CONTEXT pgContext)
{
```

Reply:
Regarding SAL, the little SAL this file uses is v0 (what is the name of IN, OUT SAL?) but OK...